src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/BuildTest.cs

```diff
@@ -1571,19 +1566,48 @@ public void BuildApplicationWithJavaSource (bool isRelease, bool expectedResult)
 		{
-			var proj = new XamarinAndroidApplicationProject () {
+			var lib = new XamarinAndroidLibraryProject {
+				ProjectName = "Library1",
 				IsRelease = true,
 				OtherBuildItems = { new AndroidItem.AndroidEnvironment ("Mono.env") {
 						TextContent = () => "MONO_DEBUG=soft-breakpoints"
 					},
 				},
 			};
-			proj.SetProperty ("_AndroidSequencePointsMode", sequencePointsMode);
-			using (var b = CreateApkBuilder (Path.Combine ("temp", TestName))) {
-				b.Verbosity = LoggerVerbosity.Diagnostic;
-				Assert.IsTrue (b.Build (proj), "Build should have succeeded.");
-				var apk = Path.Combine (Root, b.ProjectDirectory,
-					proj.IntermediateOutputPath, "android", "bin", "UnnamedProject.UnnamedProject.apk");
+			var app = new XamarinAndroidApplicationProject () {
+				IsRelease = true,
+				AndroidLinkModeRelease = AndroidLinkMode.Full,
+				References = {
+					new BuildItem ("ProjectReference","..\\Library1\\Library1.csproj"),
+				},
+			};
+			app.MainActivity = app.DefaultMainActivity.Replace ("public class MainActivity : Activity", "public class MainActivity : Xamarin.Forms.Platform.Android.FormsAppCompatActivity");
+			app.Packages.Add (KnownPackages.XamarinForms_3_0_0_561731);
+			app.Packages.Add (KnownPackages.Android_Arch_Core_Common_26_1_0);
+			app.Packages.Add (KnownPackages.Android_Arch_Lifecycle_Common_26_1_0);
+			app.Packages.Add (KnownPackages.Android_Arch_Lifecycle_Runtime_26_1_0);
+			app.Packages.Add (KnownPackages.AndroidSupportV4_27_0_2_1);
+			app.Packages.Add (KnownPackages.SupportCompat_27_0_2_1);
+			app.Packages.Add (KnownPackages.SupportCoreUI_27_0_2_1);
+			app.Packages.Add (KnownPackages.SupportCoreUtils_27_0_2_1);
+			app.Packages.Add (KnownPackages.SupportDesign_27_0_2_1);
+			app.Packages.Add (KnownPackages.SupportFragment_27_0_2_1);
+			app.Packages.Add (KnownPackages.SupportMediaCompat_27_0_2_1);
+			app.Packages.Add (KnownPackages.SupportV7AppCompat_27_0_2_1);
+			app.Packages.Add (KnownPackages.SupportV7CardView_27_0_2_1);
+			app.Packages.Add (KnownPackages.SupportV7MediaRouter_27_0_2_1);
+			app.Packages.Add (KnownPackages.SupportV7RecyclerView_27_0_2_1);
+			//LinkSkip one assembly that contains __AndroidLibraryProjects__.zip
+			string linkSkip = KnownPackages.SupportV7AppCompat_27_0_2_1.Id;
+			app.SetProperty ("AndroidLinkSkip", linkSkip);
+			app.SetProperty ("_AndroidSequencePointsMode", sequencePointsMode);
+			using (var libb = CreateDllBuilder (Path.Combine ("temp", TestName, lib.ProjectName)))
+			using (var appb = CreateApkBuilder (Path.Combine ("temp", TestName, app.ProjectName))) {
+				Assert.IsTrue (libb.Build (lib), "Library build should have succeeded.");
+				Assert.IsTrue (appb.Build (app), "App should have succeeded.");
+				Assert.IsTrue (StringAssertEx.ContainsText (appb.LastBuildOutput, $"Save assembly: {linkSkip}"), $"{linkSkip} should be saved, and not linked!");
```

Comment:
@radekdoulik when I tried this, this assembly was set to `Save` so it was working as expected?

I tried both `AndroidLinkMode.Full` and `AndroidLinkMode.SdkOnly`. Would the `StripEmbeddedLibraries` step not stripping `Skip` assemblies only be a problem if this linker step was used on a `Debug` build? (It's not used in `Debug` right now)

Reply:
I think the check for `Save` here should be good enough. So hopefully we don't need to check for the method itself.